packages/peregrine/lib/talons/SignIn/useSignIn.js

```diff
@@ -54,12 +57,30 @@ export const useSignIn = props => {
                     response && response.data.generateCustomerToken.token;
 
                 await setToken(token);
-                await getUserDetails({ fetchUserDetails });
 
                 // Then remove the old, guest cart and get the cart id from gql.
                 // TODO: This logic may be replacable with mergeCart in 2.3.4
                 await removeCart();
 
+                // Pause automatic persistence.
+                client.persistor.pause();
+
+                // Delete everything in the storage provider.
+                await client.persistor.purge();
+
+                // Delete any cached cart data.
+                Object.keys(client.cache.data.data).forEach(key => {
+                    if (key.match(/^Cart/)) {
+                        // Delete the cache entry
+                        client.cache.data.delete(key);
+
+                        // TODO: Delete the root query.
```

Comment:
If we care about deleting old cart ids/data on sign in then we will need to figure out some way to do this. Iterating over the keys for cart entries works to delete the actual data but the query itself remains in the `ROOT_QUERY` object. I'm not sure how relevant this. See below images:

Before sign in:

[![Image from Gyazo](https://i.gyazo.com/a876736a31856c96c51d4c740ebc0378.png)](https://gyazo.com/a876736a31856c96c51d4c740ebc0378)

After sign in:

[![Image from Gyazo](https://i.gyazo.com/3778805c9df5d31383d13f825183fea1.png)](https://gyazo.com/3778805c9df5d31383d13f825183fea1)

Notice how after sign in the cache entry is gone but the `ROOT_QUERY` entry remains for the old cart.

Reply:
I was able to resolve this by using `@connection` directive on EVERY query and mutation that uses cart id :o